engine/runtime/src/main/java/org/enso/interpreter/runtime/error/WithWarnings.java

```diff
@@ -53,6 +53,7 @@ private WithWarnings(Object value, int maxWarnings, EconomicSet<Warning> warning
     this(value, maxWarnings, warnings, false, additionalWarnings);
   }
 
+  @CompilerDirectives.TruffleBoundary
   private static int warningsLimitFromContext(EnsoContext ctx) {
     return ctx.getEnvironment().getOptions().get(RuntimeOptions.WARNINGS_LIMIT_KEY);
```

Comment:
I think the value of `ctx.getEnvironment().getOptions().get(RuntimeOptions.WARNINGS_LIMIT_KEY);` should be read when `EnsoContext` is created and stored in `EnsoContext`'s `final` field.